src/gridstack.scss

```diff
@@ -88,20 +88,8 @@ $animation_speed: .3s !default;
       }
     }
 
-    @for $i from 0 through $gridstack-columns {
-      &[gs-w='#{$i}'] { width: math.div(100%, $gridstack-columns) * $i; }
-      &[gs-x='#{$i}'] { left: math.div(100%, $gridstack-columns) * $i; }
-      &[gs-min-w='#{$i}'] { min-width: math.div(100%, $gridstack-columns) * $i; }
-      &[gs-max-w='#{$i}'] { max-width: math.div(100%, $gridstack-columns) * $i; }
-    }
-  }
-
-  &.grid-stack-1>.grid-stack-item {
-    min-width: 100%;
-    &[gs-w='1'] { width: 100%; }
-    &[gs-x='1'] { left: 100%; }
-    &[gs-min-w='1'] { min-width: 100%; }
-    &[gs-max-w='1'] { max-width: 100%; }
+    width: calc(calc(100% / var(--grid-column)) * var(--gs-w));
```

Comment:
why do you have this ? we don't set width or left and nothing column based should be here with your cxhanges anyway,  line 8 $gridstack-columns: should be removed.

Reply:
also if you are using dynamic style var(--grid-column) there is a performance tradeoff vs having it pre calculated. might be an issue with many grid items....

Reply:
I think the performance overhead will minimal and would not be noticable, even for large number of items.

Reply:
What do you mean by "nothing column based should not be there"? We should recompute grid items width and left position based on the updated column right? That's why I used the var(--grid-column), I set the attribute in the grid-container's div element and that will be recomputed here.

Reply:
ok I see now. but it's not ever just 'width' but gs-w 1/2/3/4/5/... you are assuming items are always 1x1 ?